### PR TITLE
Inherit from dbt-spark for backward compatibility

### LINKED ISSUE #32: dbt-databricks adapter plugin  inherits from the dbt-spark plugin
### Describe the feature

Teams using dbt-spark and dbt-labs/spark-utils intending to migrate to using dbt-databricks cant do so because adapter name is now `databricks` instead of spark, so macros that are prefixed with `spark__` won't automatically be picked up for use.

### Describe alternatives you've considered

The dbt-databricks adapter plugin depends on / inherits from the dbt-spark plugin. This would have the effect of telling dbt, "For adapter-dispatched macros, look for databricks__, then spark__, then default__

### Who will this benefit?
What kind of use case will this feature be useful for? Please be specific and provide examples, this will help us prioritize properly.

teams wanting to use dbt-databricks with dbt-labs/spark-utils macros

### PR DESCRIPTION
**Pull request by superdupershant**
Resolves #32
Fixes dbt-labs/spark-utils#23

Integration tests will be added to dbt-labs/spark_utils to ensure compatibility with dbt-databricks.

### Description

Inherit from dbt-spark in order to maintain compatibility with existing spark-utils and other macros.

### Checklist

- [ x] I have run this code in development and it appears to resolve the stated issue

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 databricks-sql-connector>=0.9.0
+dbt-spark ~= 1.0.0
```
Comment (ueshin) [inline requirements.txt:2]: Just a nit: maybe we want the definition format to be consistent.

```
databricks-sql-connector >= 0.9.0
dbt-spark ~= 1.0.0
```

or

```
databricks-sql-connector>=0.9.0
dbt-spark~=1.0.0
```

**Review by ueshin — COMMENTED** [review]
Otherwise, LGTM.
Let me try something with this change in my local as well.

**Review by jtcohen6 — APPROVED** [review]
Looks right to me!

This change also enables you to remove methods + macros that are duplicated from / identical to those in `dbt-spark`, as you're able/interested, or inherit new functionality going forward. Let's keep talking and thinking through the right separation of concerns

**Review by ueshin — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**